### PR TITLE
fix serde not compiling for struct with lifetime

### PR DESCRIPTION
**Pull request by b1ek**

### CODE REVIEW

**Comment by Ph0enixKM** [comment]
Provided that the code variable contains a long script file (potentially 1 mln lines of code), this could lead to unnecessary value cloning. Of course, as of the alpha version, this won't be too much of an issue. Let's consider removing serde's serialise and deserialise for `Reader` as it's only used internally, and importing it publicly from the package should be considered a violation of encapsulation rules.

File: src/compiling/lexing/reader.rs [hunk 1]
```diff
@@ -10,18 +10,18 @@ pub enum ReadMode {
 }
 
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```
Comment (Ph0enixKM) [inline src/compiling/lexing/reader.rs:12]: I'd just remove the serde's serialize and deserialize

File: src/compiling/lexing/reader.rs [hunk 2]
```diff
@@ -10,18 +10,18 @@ pub enum ReadMode {
...
-pub struct Reader<'a> {
-    pub code: &'a String,
+pub struct Reader {
```
Comment (Ph0enixKM) [inline src/compiling/lexing/reader.rs:13]: And I'd rename `pub` to `pub(crate)` instead

**Review by Ph0enixKM — CHANGES_REQUESTED** [review]
Provided that the code variable contains a long script file (potentially 1 mln lines of code), this could lead to unnecessary value cloning. Of course, as of the alpha version, this won't be too much of an issue. Let's consider removing serde's serialise and deserialise for Reader as it's only used internally, and importing it publicly from the package should be considered a violation of encapsulation rules.

File: src/compiling/lexing/reader.rs (hunk 1 above)
Comment (b1ek) [reply src/compiling/lexing/reader.rs:12]: im not sure if there is a way to remove it without reverting #66

Comment (b1ek) [reply src/compiling/lexing/reader.rs:12]: im not sure if there is a way to do that without reverting #66

File: src/compiling/lexing/reader.rs (hunk 2 above)
Comment (b1ek) [reply src/compiling/lexing/reader.rs:8]: what for?

Comment (Ph0enixKM) [reply src/compiling/lexing/reader.rs:8]: To prevent users from importing this structure out of an external module. This could be done in a separate issue

**Review by Ph0enixKM — CHANGES_REQUESTED** [review]

**Review by Ph0enixKM — APPROVED** [review]

**Pull request merged**